elements/radio-group/radio-group.js

```diff
@@ -1,151 +1,233 @@
-// `RadioButton` is the radio button element itself. Maybe surprisingly,
-// there’s almost no logic on this element. 
-class RadioButton extends HTMLElement {
-  constructor() {
-    super();
-  }
+/**
+ * A `RadioGroup` is a set checkable buttons, where only one button may be
+ * checked at a time. The `RadioGroup` element wraps a set of `RadioButton`
+ * children and manages their checked states in response to user keyboard
+ * actions such as pressing arrow keys to select the next radio button, or if
+ * the user clicks with a mouse.
+ */
+
+(function() {
+  /**
+   * Define keycodes to help with handling keyboard events.
+   */
+  const KEYCODE = {
+    DOWN: 40,
+    LEFT: 37,
+    RIGHT: 39,
+    SPACE: 32,
+    UP: 38
+  };
+
+  /**
+   * `RadioButton` is a simple, checkable button.
+   */
+  class RadioButton extends HTMLElement {
+    constructor() {
+      super();
+    }
 
-  connectedCallback() {
-    // All the juggling of ARIA attributes will be handled 
-    // in the surrounding `RadioGroup`. We only initialize the
-    // the radio button when it gets attached to the DOM.
-    this.setAttribute('role', 'radio');
-    this.setAttribute('tabindex', -1);
-    this.setAttribute('aria-checked', false);
-  }
-}
-
-window.customElements.define('radio-button', RadioButton);
-
-// For interop reasons[1], we’ll have to maintain our map
-// for semantic key names.
-const KEYCODE = {
-  DOWN: 40,
-  LEFT: 37,
-  RIGHT: 39,
-  SPACE: 32,
-  UP: 38
-};
-
-// The `RadioGroup` does all the heavy lifting. It attaches
-// event listeners, adjust ARIA attribute values and provides
-// the API for this element.
-class RadioGroup extends HTMLElement {
-  constructor() {
-    super();
+    /**
+     * The `RadioButton` sets its initial ARIA attributes when it's attached to
+     * the DOM. The surrounding RadioGroup handles dynamic changes to the ARIA
+     * attributes. The RadioButton should always set a `role` of `radio`, and
+     * should check to see if its `tabindex` and `aria-checked` values have been
+     * set by the user. Otherwise, it should use default values.
+     */
+    connectedCallback() {
+      this.setAttribute('role', 'radio');
+      this.setAttribute('tabindex', this.getAttribute('tabindex') || -1);
```

Comment:
I guess i don't have a great reason except for consistency (eventually it may get set to -1 as part of the checked/unchecked process). I could remove this line entirely if you think that might be preferable.